### DISCUSSION TITLE
Add annotations

### DISCUSSION THREAD

**Discussion by BeritJanssen** [UUDigitalHumanitieslab/I-analyzer#1424] [Ideas]
Next to tagging documents, it is also conceivable that users may want to annotate them.

**Comment by lukavdplas** [comment]
Question: making annotation is possible in ontraqh, would it be possible to transfer this to I-analyer? @JeltevanBoheemen @ar-jan

**Comment by JeltevanBoheemen** [comment]
I would call the Ontraqh implementation rickety at best. 
The code for selecting text works fine, can reuse some of it.
The code for rendering annotations (and links between annotations) is a mess of `ngZone`, `setTimeout` and events going up and down 4 levels of `Input/Output`. 

So it could be transfered, with a whole lot of refactoring.

**Comment by ar-jan** [comment]
We also experienced poor rendering performance in some situations which we didn't fully resolve.

**Comment by lukavdplas** [comment]
Out of curiosity, how did you save annotations in the backend? In the SQL database or elasticsearch?

**Comment by ar-jan** [comment]
In the SQL database. We did consider ES, but that would've involved working around the lack of standard MySQL features, mainly referential integrity for storing links between annotations, and user account creating the annotation.

**Comment by lukavdplas** [comment]
For the record: we're currently looking into adding integration with the [scholarly web annotation client](https://clariah.github.io/scholarly-web-annotation-client/)
